ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerSubscription.scala

```diff
@@ -141,9 +136,44 @@ object ParallelIndexerSubscription {
           logger.info(s"Storing ${update.description}")
       }
     }
-    val batch = input.iterator.flatMap { case ((offset, update), _) =>
+
+    val nonMetering = input.iterator.flatMap { case ((offset, update), _) =>
       toDbDto(offset)(update)
     }.toVector
+
+    val metering = input.iterator
+      .collect({ case ((offset, ta: TransactionAccepted), time) =>
+        (offset, ta.optCompletionInfo, time)
+      })
+      .collect({
+        case (offset, Some(CompletionInfo(_, applicationId, _, _, _, Some(statistics))), time) =>
+          (offset, applicationId, statistics, time)
+      })
+      .foldLeft(Map.empty[Ref.ApplicationId, DbDto.TransactionMetering]) {
+        case (map, (offset, applicationId, statistics, time)) =>
+          val update = map.get(applicationId) match {
+            case None =>
+              DbDto.TransactionMetering(
+                applicationId,
+                statistics.committed.actions + statistics.rolledBack.actions,
+                time,
```

Comment:
Yes, we must use microseconds for timestamps, as otherwise the year `9999` does not fit into 63 bits.

Also: consider what happens when the indexer goes through a HA recovery and switches the process on which indexing happens. Are we relying here on the timestamps on inserts being monotonic? If yes, then that needs to be guaranteed by storing the last timestamp in the parameters table and using it as a lower-bound for the measured timestamp.

Reply:
Will switch to `Timestamp.now()`

Reply:
Done.